relay-server/src/services/buffer/envelope_stack/sqlite.rs

```diff
@@ -183,11 +180,10 @@ impl EnvelopeStack for SqliteEnvelopeStack {
         Ok(Some(envelope))
     }
 
-    fn flush(self) -> Vec<Box<Envelope>> {
-        self.batch
-            .into_iter()
-            .filter_map(|e| e.try_into().ok())
-            .collect()
+    async fn flush(mut self) {
```

Comment:
I understand your change's rationale, but isn't it better if we also batch the batch inserts? This way we will emit multiple queries for each stack.

Reply:
It's at most one db query per stack. It would be more optimal to batch small stacks together (not into the same row but in a batch SQL statement), but I did not want to add that complexity for now.

We'll have to measure the flush time, if it takes too long we can optimize.